[release][jobs] jobs_specify_num_gpus is failing

https://console.anyscale-staging.com/o/anyscale-internal/projects/prj_qC3ZfndQWYYjx2cz8KWGNUL4/clusters/ses_irux6zc5trsu1cvc1qg8fr6y?command-history-section=command_history

```
Traceback (most recent call last):
  File "jobs_check_cuda_available.py", line 22, in <module>
    assert torch.cuda.is_available(), "CUDA is not available in the driver script"
AssertionError: CUDA is not available in the driver script

status: PENDING
status: PENDING
status: RUNNING
status: RUNNING
status: RUNNING
status: RUNNING
status: RUNNING
status: RUNNING
status: RUNNING
status: RUNNING
status: RUNNING
status: RUNNING
status: RUNNING
status: RUNNING
status: RUNNING
status: RUNNING
status: RUNNING
status: RUNNING
status: RUNNING
status: FAILED
Status message:  Job failed due to an application error, last available logs (truncated to 20,000 chars):
2022-11-28 09:29:04,511 INFO worker.py:1352 -- Connecting to existing Ray cluster at address: 172.31.244.165:9031...
2022-11-28 09:29:04,517 INFO worker.py:1535 -- Connected to Ray cluster. View the dashboard at https://console.anyscale-staging.com/api/v2/sessions/ses_irux6zc5trsu1cvc1qg8fr6y/services?redirect_to=dashboard 
(scheduler +4s, ip=172.31.244.165) Tip: use `ray status` to view detailed cluster status. To disable these messages, set RAY_SCHEDULER_EVENTS=0.
(scheduler +4s, ip=172.31.244.165) Resized to 32 CPUs, 1 GPUs.
Traceback (most recent call last):
  File "jobs_check_cuda_available.py", line 51, in <module>
    wait_for_condition(check_num_nodes_and_spawn_tasks)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/test_utils.py", line 486, in wait_for_condition
    raise RuntimeError(message)
RuntimeError: The condition wasn't met before the timeout expired.

Traceback (most recent call last):
  File "workloads/jobs_specify_num_gpus.py", line 84, in <module>
    assert status == JobStatus.SUCCEEDED
AssertionError
```

--- a/release/jobs_tests/workloads/jobs_check_cuda_available.py
+++ b/release/jobs_tests/workloads/jobs_check_cuda_available.py
@@ -35,7 +35,7 @@ assert ray.get(f.remote()) == [0]
 NUM_NODES = 2
 
 
-@ray.remote(num_cpus=1)
+@ray.remote(num_cpus=1, scheduling_strategy="SPREAD")
 def get_node_id():
     return ray.get_runtime_context().node_id
 

